docs/UserMenu.md

```diff
@@ -0,0 +1,107 @@
+# User Menu
+
+The `<pxb-user-menu>` is an Avatar that opens a Menu when clicked. It is typically used in the top-right corner of an application and indicates who is logged in.
+
+<div style="align-items: center; display:flex; justify-content: space-around">
+
+<img width="30%" alt="UserMenu Avatar" src="./images/userMenuAvatar.png">
+<img width="35%" alt="UserMenu Opened" src="./images/userMenuOpened.png">
+
+</div>
+
+The Menu can be populated via the `menuGroups` prop, or can be entirely customized by supplying your own content.  See the API section below for more details.
+
+## Usage
+
+<div style="width: 100%; text-align: center">
+    <img width="100%" style="max-width: 500px" alt="User Menu Anatomy" src="./images/userMenuAnatomy.png">
+</div>
+
+```typescript
+// app.module.ts
+import { UserMenuModule, InfoListItemModule } from '@pxblue/angular-components';
+...
+imports: [
+    UserMenuModule,
+    InfoListItemModule
+],
+...
+```
+
+```tsx
+const items = [
+    {
+        title: 'Account',
+        icon: 'settings',
+    },
+    {
+        title: 'Contact Us',
+        icon: 'mail',
+    },
+    {
+        title: 'Log Out',
+        icon: 'logout',
+    }
+];
+```
+
+```html
+<pxb-user-menu avatarValue="HA" menuTitle='Sample Title '[(open)]="state.open">
+    <mat-nav-list pxb-body [style.paddingTop.px]="0">
+        <pxb-info-list-item *ngFor="let item of items" [dense]="true">
+            <mat-icon pxb-icon>{{item.icon}}</mat-icon>
+            <div pxb-title>{{item.title}}</div>
+        </pxb-info-list-item>
+    </mat-nav-list>
+</pxb-user-menu> 
+```
+
+## API
+
+Parent element (`<pxb-user-menu>`) attributes:
+
+<div style="overflow: auto;">
+
+| @Input       | Description                                      | Type                    | Required | Default |
+| ------------ | ------------------------------------------------ | ----------------------- | -------- | ------- |
+| avatarImage  | Image source for avatar                          | `string`                | no       |         |
+| avatarValue  | Text value for avatar                            | `string`                | no       |         |
```

Comment:
```
<div class="pxb-user-menu-avatar" [class.non-text-avatar]="!avatarValue">
            <ng-container *ngIf="avatarValue">{{ avatarValue }}</ng-container>
            <img *ngIf="avatarImage" [src]="avatarImage" alt="User Menu Avatar" />
            <ng-content></ng-content>
        </div>
```

It will try to render both inside the small avatar window.

Reply:
Same goes for if they also provide a custom avatar content too, then it'll try to render all 3 at once.

Reply:
At some point we may want to think about going in and having a preference on things like this (e.g., if custom avatar, render than and nothing else, if no custom avatar, check if theres a image avatar, if not then look for value) so that a user doesn't accidentally run into the situation where they've got 3 avatars controlling the menu. On quick glance it doesn't look like we're doing this anywhere else either (Drawer Header, Navgroup header, etc.), so it should be alright for now.

Reply:
That's something for docit design overhaul, probably not for our api